### PR TITLE
Greatly increase test coverage for pkg/sign

### PR DESCRIPTION
**Pull request by steiza**
#### Summary

As we work towards signing support (#136) we want to ensure there's good test coverage of the code, in addition to conformance tests. This increases the built-in test coverage (e.g. not conformance tests) from ~23% to ~65%.

#### Release Note

NONE - we aren't ready to release signing yet

#### Documentation

N/A

### CODE REVIEW

File: pkg/sign/certificate.go [hunk 1]
```diff
@@ -42,6 +46,8 @@ type FulcioOptions struct {
 	Retries uint
 	// Optional version string for user agent
 	LibraryVersion string
+	// Optional HTTP client (for dependency injection)
+	Client HTTPClient
```
Comment (codysoyland) [inline pkg/sign/certificate.go]: Nit: I suggest using [`http.RoundTripper`](https://pkg.go.dev/net/http#RoundTripper) for dependency injection.

File: pkg/sign/certificate_test.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"github.com/sigstore/sigstore-go/pkg/testing/ca"
+)
+
+var virtualSigstore *ca.VirtualSigstore
+
+func getFulcioResponse() (*http.Response, error) {
+	var err error
+	if virtualSigstore == nil {
```
Comment (codysoyland) [inline pkg/sign/certificate_test.go:37]: Mutating a package level var can result in bugs if we go add another test that uses this same var (as tests can run in parallel), so I would guard this check with [`sync.Once`](https://pkg.go.dev/sync#Once.Do), which is like a single-use mutex.

File: pkg/sign/certificate.go (hunk 1 above)
Comment (Hayden-IO) [reply pkg/sign/certificate.go]: I was reading a bit about the idiomatic way to do testing in Golang, and a common [suggestion](https://medium.com/zus-health/mocking-outbound-http-requests-in-go-youre-probably-doing-it-wrong-60373a38d2aa#:~:text=There%20is%20an%20easier%2C%20idiomatic%20way) was https://pkg.go.dev/net/http/httptest

File: pkg/sign/certificate.go [hunk 3]
```diff
@@ -75,6 +81,14 @@ type chain struct {
 }
 
 func NewFulcio(opts *FulcioOptions) *Fulcio {
+	if opts.Client == nil {
+		client := &http.Client{}
+		if opts.Timeout != 0 {
```
Comment (kommendorkapten) [inline pkg/sign/certificate.go]: If no timeout is provided I would still configure a timeout (30s?), as the default client blocks "forever" and that can lead to very unexpected behaviours.

File: pkg/sign/certificate.go (hunk 1 above)
Comment (codysoyland) [reply pkg/sign/certificate.go]: Interesting, I've used the pkg httptest for testing handlers before, but not outbound requests. I think for this usecase, I still prefer replacing the transport with a custom `RoundTripper` as it seems simpler and doesn't open a port in tests.

**Review by codysoyland — DISMISSED** [review]
nice work!

**Review by Hayden-IO — DISMISSED** [review]

**Review by Hayden-IO — APPROVED** [review]

**Review by codysoyland — APPROVED** [review]

**Pull request merged**